File: src/main/java/com/server/ordering/domain/dto/response/OrderPreviewDto.java
package com.server.ordering.domain.dto.response;

import com.server.ordering.domain.Order;
import com.server.ordering.domain.OrderStatus;
import com.server.ordering.domain.OrderType;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.format.DateTimeFormatter;

import static lombok.AccessLevel.PROTECTED;

@Getter @Setter
@AllArgsConstructor
@NoArgsConstructor(access = PROTECTED)
public class OrderPreviewDto {

    private Long orderId;
    private Long reviewId;
    private Integer myOrderNumber;
    private String orderSummary;
    private String receivedTime;
    private String checkTime;
    private String cancelOrCompletedTime;
    private int totalPrice;
    private OrderType orderType;
    private Integer tableNumber;
    private OrderStatus orderStatus;

    public OrderPreviewDto(Order order) {
        this.orderId = order.getId();
        this.reviewId = order.getReview() == null ? null : order.getReview().getId();
        this.myOrderNumber = order.getMyOrderNumber();
        this.orderSummary = order.getOrderSummary();
        this.receivedTime = order.getReceivedTime().format(DateTimeFormatter.ofPattern("yyyy/MM/dd - HH:mm:ss"));
        this.checkTime = order.getCheckedTime() == null ? null
                : order.getCheckedTime().format(DateTimeFormatter.ofPattern("yyyy/MM/dd - HH:mm:ss"));
        this.cancelOrCompletedTime = order.getCanceledOrCompletedTime() == null ? null
                : order.getCanceledOrCompletedTime().format(DateTimeFormatter.ofPattern("yyyy/MM/dd - HH:mm:ss"));
        this.totalPrice = order.getTotalPrice();
        this.orderType = order.getOrderType();
        this.tableNumber = order.getTableNumber();
        this.orderStatus = order.getStatus();
    }
}

File: src/main/kotlin/com/byeoru/ordering_server/domain/dto/response/OrderPreviewDto.kt
package com.byeoru.ordering_server.domain.dto.response

import com.byeoru.ordering_server.domain.Order
import com.byeoru.ordering_server.domain.OrderStatus
import com.byeoru.ordering_server.domain.OrderType
import java.time.format.DateTimeFormatter

open class OrderPreviewDto(order: Order) {

    val orderId: Long?
    val reviewId: Long?
    val myOrderNumber: Int?
    val orderSummary: String?
    val receivedTime: String?
    val checkTime: String?
    val cancelOrCompletedTime: String?
    val totalPrice: Int
    val orderType: OrderType
    val tableNumber: Int?
    val orderStatus: OrderStatus?

    init {
        this.orderId = order.id
        this.reviewId = order.review?.id
        this.myOrderNumber = order.myOrderNumber
        this.orderSummary = order.orderSummary
        this.receivedTime = order.receivedTime?.format(DateTimeFormatter.ofPattern("yyyy/MM/dd - HH:mm:ss"))
        this.checkTime = order.checkedTime?.format(DateTimeFormatter.ofPattern("yyyy/MM/dd - HH:mm:ss"))
        this.cancelOrCompletedTime = order.canceledOrCompletedTime?.format(DateTimeFormatter.ofPattern("yyyy/MM/dd - HH:mm:ss"))
        this.totalPrice = order.totalPrice
        this.orderType = order.orderType
        this.tableNumber = order.tableNumber
        this.orderStatus = order.status
    }
}